Error loading `inertia-svelte` under Rails (via webpacker)

```
8:34:16 PM webpack.1 |  Version: webpack 4.44.2
8:34:16 PM webpack.1 |  Time: 3899ms
8:34:16 PM webpack.1 |  Built at: 10/25/2020 8:34:15 PM
8:34:16 PM webpack.1 |                                                     Asset       Size       Chunks                         Chunk Names
8:34:16 PM webpack.1 |                       application-141aa79e56d42feb82ca.js    986 KiB  application  [emitted] [immutable]  application
8:34:16 PM webpack.1 |                   application-141aa79e56d42feb82ca.js.map   1.02 MiB  application  [emitted] [dev]        application
8:34:16 PM webpack.1 |                            client-141aa79e56d42feb82ca.js    654 KiB       client  [emitted] [immutable]  client
8:34:16 PM webpack.1 |                        client-141aa79e56d42feb82ca.js.map    674 KiB       client  [emitted] [dev]        client
8:34:16 PM webpack.1 |                                             manifest.json  832 bytes               [emitted]              
8:34:16 PM webpack.1 |  media/images/banner-5f539d39241aceb17c07bbaa6e611cac.svg   16.2 KiB               [emitted]              
8:34:16 PM webpack.1 |    media/images/icon-03960d2ed0f0c47a77a9fbf21ddf798f.svg   2.71 KiB               [emitted]              
8:34:16 PM webpack.1 |  ERROR in ./node_modules/@inertiajs/inertia-svelte/src/index.js
8:34:16 PM webpack.1 |  Module not found: Error: [CaseSensitivePathsPlugin] `/Users/jason/Projects/gateway/node_modules/@inertiajs/inertia-svelte/src/link.svelte` does not match the corresponding path on disk `Link.svelte`.
8:34:16 PM webpack.1 |  ℹ ｢wdm｣: Failed to compile.
```

I followed the instructions in the Inertia docs, and couldn't get past this issue. I believe it has to do with webpack being confused while loading both `Link.svelte` and `link.js`. My app has a fairly vanilla setup, but I will try to reproduce it in a empty rails app.

--- a/packages/inertia-svelte/src/App.svelte
+++ b/packages/inertia-svelte/src/App.svelte
@@ -1,7 +1,7 @@
 <script>
   import { Inertia } from '@inertiajs/inertia'
-  import Render from './Render.svelte'
   import store from './store'
+  import Render, { h } from './Render.svelte'
 
   export let initialPage, resolveComponent, resolveErrors, transformProps
 
@@ -13,22 +13,22 @@
     swapComponent: async ({ component, page, preserveState }) => {
       store.update((current) => ({
         component,
-        layout: (() => {
-          if (!component.layout) {
-            return []
-          } else if (Array.isArray(component.layout)) {
-            return component.layout
-          } else {
-            return [component.layout]
-          }
-        })(),
         page,
         key: preserveState ? current.key : Date.now()
       }))
     }
   })
+
+  $: child = $store.component && h($store.component.default, $store.page.props)
+  $: layout = $store.component && $store.component.layout
+  $: components = layout
+    ? Array.isArray(layout)
+      ? layout
+          .concat(child)
+          .reverse()
+          .reduce((child, layout) => h(layout, {}, [child]))
+      : h(layout, {}, [child])
+    : child
 </script>
 
-{#if $store.component}
-  <Render />
-{/if}
+<Render {...components} />

--- a/packages/inertia-svelte/src/App.svelte
+++ b/packages/inertia-svelte/src/App.svelte
@@ -1,12 +1,9 @@
 <script>
   import { Inertia } from '@inertiajs/inertia'
+  import Render from './Render.svelte'
   import store from './store'
 
-  export let
-    initialPage,
-    resolveComponent,
-    resolveErrors,
-    transformProps
+  export let initialPage, resolveComponent, resolveErrors, transformProps
 
   Inertia.init({
     initialPage,
@@ -14,69 +11,24 @@
     resolveErrors,
     transformProps,
     swapComponent: async ({ component, page, preserveState }) => {
-      store.update(current => ({
+      store.update((current) => ({
         component,
         layout: (() => {
           if (!component.layout) {
             return []
           } else if (Array.isArray(component.layout)) {
-            if (component.layout.length > 5) {
-              throw new Error('The Svelte adapter only supports up to five nested layouts.')
-            }
             return component.layout
           } else {
             return [component.layout]
           }
         })(),
         page,
-        key: preserveState ? current.key : Date.now(),
+        key: preserveState ? current.key : Date.now()
       }))
-    },
+    }
   })
 </script>
 
 {#if $store.component}
-  {#if $store.layout.length > 0}
-    <svelte:component this={$store.layout[0]}>
-      {#if $store.layout.length > 1}
-        <svelte:component this={$store.layout[1]}>
-          {#if $store.layout.length > 2}
-            <svelte:component this={$store.layout[2]}>
-              {#if $store.layout.length > 3}
-                <svelte:component this={$store.layout[3]}>
-                  {#if $store.layout.length > 4}
-                    <svelte:component this={$store.layout[4]}>
-                      {#key $store.key}
-                        <svelte:component this={$store.component.default} {...$store.page.props} />
-                      {/key}
-                    </svelte:component>
-                  {:else}
-                    {#key $store.key}
-                      <svelte:component this={$store.component.default} {...$store.page.props} />
-                    {/key}
-                  {/if}
-                </svelte:component>
-              {:else}
-                {#key $store.key}
-                  <svelte:component this={$store.component.default} {...$store.page.props} />
-                {/key}
-              {/if}
-            </svelte:component>
-          {:else}
-            {#key $store.key}
-              <svelte:component this={$store.component.default} {...$store.page.props} />
-            {/key}
-          {/if}
-        </svelte:component>
-      {:else}
-        {#key $store.key}
-          <svelte:component this={$store.component.default} {...$store.page.props} />
-        {/key}
-      {/if}
-    </svelte:component>
-  {:else}
-    {#key $store.key}
-      <svelte:component this={$store.component.default} {...$store.page.props} />
-    {/key}
-  {/if}
+  <Render />
 {/if}

--- a/packages/inertia-svelte/src/Render.svelte
+++ b/packages/inertia-svelte/src/Render.svelte
@@ -18,7 +18,7 @@
 
 {#if $store.component}
   <svelte:component this={component} {...props}>
-    {#each children as child}
+    {#each children as child ($store.key)}
       <svelte:self {...child} />
     {/each}
   </svelte:component>

--- a/packages/inertia-svelte/src/Render.svelte
+++ b/packages/inertia-svelte/src/Render.svelte
@@ -1,14 +1,25 @@
+<script context="module">
+  export const h = (component, props, children) => {
+    return {
+      component,
+      ...(props ? { props } : {}),
+      ...(children ? { children } : {})
+    }
+  }
+</script>
+
 <script>
   import store from './store'
 
-  export let index = 0
-  $: component = $store.layout[index]
-    ? $store.layout[index]
-    : $store.component.default
+  export let component
+  export let props = {}
+  export let children = []
 </script>
 
-<svelte:component this={component} {...$store.page.props}>
-  {#each $store.layout as _, childIndex ($store.key)}
-    <svelte:self index={childIndex + 1} />
-  {/each}
-</svelte:component>
+{#if $store.component}
+  <svelte:component this={component} {...props}>
+    {#each children as child}
+      <svelte:self {...child} />
+    {/each}
+  </svelte:component>
+{/if}

--- a/packages/inertia-svelte/src/Render.svelte
+++ b/packages/inertia-svelte/src/Render.svelte
@@ -7,10 +7,8 @@
     : $store.component.default
 </script>
 
-{#key $store.key}
-  <svelte:component this={component} {...$store.page.props}>
-    {#each $store.layout as layout, childIndex}
-      <svelte:self {layout} index={childIndex + 1} />
-    {/each}
-  </svelte:component>
-{/key}
+<svelte:component this={component} {...$store.page.props}>
+  {#each $store.layout as layout, childIndex ($store.key)}
+    <svelte:self {layout} index={childIndex + 1} />
+  {/each}
+</svelte:component>

--- a/packages/inertia-svelte/src/Render.svelte
+++ b/packages/inertia-svelte/src/Render.svelte
@@ -8,7 +8,7 @@
 </script>
 
 <svelte:component this={component} {...$store.page.props}>
-  {#each $store.layout as layout, childIndex ($store.key)}
-    <svelte:self {layout} index={childIndex + 1} />
+  {#each $store.layout as _, childIndex ($store.key)}
+    <svelte:self index={childIndex + 1} />
   {/each}
 </svelte:component>

--- a/packages/inertia-svelte/src/Render.svelte
+++ b/packages/inertia-svelte/src/Render.svelte
@@ -1,0 +1,16 @@
+<script>
+  import store from './store'
+
+  export let index = 0
+  $: component = $store.layout[index]
+    ? $store.layout[index]
+    : $store.component.default
+</script>
+
+{#key $store.key}
+  <svelte:component this={component} {...$store.page.props}>
+    {#each $store.layout as layout, childIndex}
+      <svelte:self {layout} index={childIndex + 1} />
+    {/each}
+  </svelte:component>
+{/key}

--- a/packages/inertia-svelte/src/Render.svelte
+++ b/packages/inertia-svelte/src/Render.svelte
@@ -18,7 +18,7 @@
 
 {#if $store.component}
   <svelte:component this={component} {...props}>
-    {#each children as child ($store.key)}
+    {#each children as child, index (component && component.length === index ? $store.key : null)}
       <svelte:self {...child} />
     {/each}
   </svelte:component>

--- a/packages/inertia-svelte/src/index.js
+++ b/packages/inertia-svelte/src/index.js
@@ -1,5 +1,5 @@
 export { default as App, default as app, default as InertiaApp } from './App.svelte'
 export { default as Link, default as link, default as InertiaLink } from './Link.svelte'
-export { default as inertia } from './link'
-export { default as page } from './page'
-export { default as remember } from './remember'
+export { default as inertia } from './link.js'
+export { default as page } from './page.js'
+export { default as remember } from './remember.js'

